Tests/UnitTests/SubscriberAttributes/PurchasesSubscriberAttributesTests.swift

```diff
@@ -218,30 +222,42 @@ class PurchasesSubscriberAttributesTests: TestCase {
 
     // MARK: Notifications
 
+    func testUpdatingOnForegroundIsThrottled() {
+        self.setupPurchases()
+
+        self.mockNotificationCenter.fireNotifications()
+
+        expect(self.mockSubscriberAttributesManager.invokedSyncAttributesForAllUsersCount) == 1
+    }
+
     func testSubscribesToForegroundNotifications() {
-        setupPurchases()
+        self.setupPurchases()
 
         expect(self.mockNotificationCenter.observers).toNot(beEmpty())
 
         expect(self.mockNotificationCenter.observers).to(containElementSatisfying {
             $0.notificationName == SystemInfo.applicationWillEnterForegroundNotification
         })
 
+        self.clock.advance(by: SystemInfo.cacheUpdateThrottleDuration + .seconds(1))
         self.mockNotificationCenter.fireNotifications()
+
         expect(self.mockSubscriberAttributesManager.invokedSyncAttributesForAllUsersCount) == 2
     }
 
     func testSubscribesToBackgroundNotifications() {
-        setupPurchases()
+        self.setupPurchases()
 
         expect(self.mockNotificationCenter.observers).toNot(beEmpty())
 
         expect(self.mockNotificationCenter.observers).to(containElementSatisfying {
             $0.notificationName == SystemInfo.applicationDidEnterBackgroundNotification
         })
 
+        let previousCount = self.mockSubscriberAttributesManager.invokedSyncAttributesForAllUsersCount
         self.mockNotificationCenter.fireNotifications()
-        expect(self.mockSubscriberAttributesManager.invokedSyncAttributesForAllUsersCount) == 2
```

Comment:
This was only 2 because `fireNotifications` was firing the `didEnterForeground` and `didEnterBackground`, and this test was detecting the double update.
Now it's `1` because it only gets updated automatically (regardless of throttling) when entering the background.